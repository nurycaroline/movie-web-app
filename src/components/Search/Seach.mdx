---
name: Search
menu: Componentes
---

import {Playground} from 'docz';
import Search from '.';

# Search

<Playground>
  <Search onChangeSearchBy={() => {}} value="" />
</Playground>
