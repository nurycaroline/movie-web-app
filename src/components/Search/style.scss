@import '../../assets/variables';

.search {
  margin: 15px;

  input {
    width: 500px;
    height: 38px;
    font-size: 14px;
    color: $primary-color;
    padding-left: 10px;
  }
}
