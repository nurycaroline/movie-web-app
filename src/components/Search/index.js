import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

const Search = ({ onChangeSearchBy, value }) => (
  <div className="search">
    <input
      type="text"
      placeholder="pesquise filmes por titulo aqui"
      value={value}
      onChange={e => onChangeSearchBy(e.currentTarget.value)}
    />
  </div>
);

Search.propTypes = {
  onChangeSearchBy: PropTypes.func,
  value: PropTypes.string,
};

Search.defaultProps = {
  onChangeSearchBy: () => {},
  value: '',
};

export default Search;
