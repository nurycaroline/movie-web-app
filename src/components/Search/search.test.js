import React from 'react';
import renderer from 'react-test-renderer';
import Search from '.';

test('Input Search', () => {
  const component = renderer.create(<Search />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
