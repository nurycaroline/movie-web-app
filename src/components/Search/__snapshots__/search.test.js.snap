// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Search 1`] = `
<div
  className="search"
>
  <input
    onChange={[Function]}
    placeholder="pesquise filmes por titulo aqui"
    type="text"
    value=""
  />
</div>
`;
